lib/term.js

```diff
@@ -1,42 +1,55 @@
 const rdf = require('@rdfjs/data-model')
 
-function term (value, type, datatype) {
-  if (datatype) {
-    datatype = rdf.namedNode(datatype.value || datatype.toString())
+function blankNode (value) {
+  if (value && typeof value !== 'string') {
+    throw new Error('unknown type')
   }
 
-  if (type) {
-    if (type === 'BlankNode') {
-      return rdf.blankNode(value && value.toString())
-    }
+  return rdf.blankNode(value && value.toString())
+}
+
+function literal (value, languageOrDatatype) {
+  if (typeof value !== 'string' && typeof value !== 'number') {
+    throw new Error('unknown type')
+  }
 
-    if (type === 'Literal') {
-      return rdf.literal(value.toString(), datatype)
-    }
+  languageOrDatatype = languageOrDatatype && (languageOrDatatype.value || languageOrDatatype.toString())
 
-    if (type === 'NamedNode') {
-      return rdf.namedNode(value.toString())
-    }
+  if (languageOrDatatype && languageOrDatatype.indexOf(':') !== -1) {
+    languageOrDatatype = rdf.namedNode(languageOrDatatype)
   }
 
-  if (!value) {
-    return undefined
+  return rdf.literal(value.toString(), languageOrDatatype)
+}
+
+function namedNode (value) {
+  if (typeof value !== 'string') {
+    throw new Error('unknown type')
```

Comment:
```suggestion
    throw new Error('Named node must be a URI string')
```

Besides, would you consider checking that the value is indeed a URI? Maybe simply "substringing" for a colon?